miniapps/blocksolvers/div_free_solver.hpp

```diff
@@ -0,0 +1,252 @@
+#include "mfem.hpp"
+#include <memory>
+
+using namespace std;
+using namespace mfem;
+
+/// Parameters for iterative solver
+struct IterSolveParameters
+{
+   int print_level = 0;
+   int max_iter = 500;
+   double abs_tol = 1e-12;
+   double rel_tol = 1e-9;
+};
+
+/// Parameters for the divergence free solver
+struct DFSParameters : IterSolveParameters
+{
+   bool verbose = false;
+   bool B_has_nullity_one = false;   // whether B has a 1D nullspace
+   bool coupled_solve = false;       // whether to solve all unknowns together
```

Comment:
What this means should be documented somewhere (maybe in `DivFreeSolver` object), it's not clearly described anywhere even though it's a command-line argument for the driver.